internal/component/beyla/ebpf/beyla_linux_test.go

```diff
@@ -178,24 +207,64 @@ func TestConvert_Discovery(t *testing.T) {
 				},
 			},
 		},
+		ExcludeServices: []Service{
+			{
+				Name:      "test",
+				Namespace: "default",
+			},
+		},
 	}
 	config, err := args.Convert()
 
 	require.NoError(t, err)
-	require.Len(t, config.Services, 1)
+	require.Len(t, config.Services, 3)
 	require.Equal(t, "test", config.Services[0].Name)
 	require.Equal(t, "default", config.Services[0].Namespace)
 	require.Equal(t, services.PortEnum{Ranges: []services.PortRange{{Start: 80, End: 0}}}, config.Services[0].OpenPorts)
-	require.True(t, config.Services[0].Path.IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrNamespace].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrPodName].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrDeploymentName].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrReplicaSetName].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrStatefulSetName].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrDaemonSetName].IsSet())
-	require.True(t, config.Services[0].Metadata[services.AttrOwnerName].IsSet())
-	require.True(t, config.Services[0].PodLabels["test"].IsSet())
+	require.True(t, config.Services[1].Metadata[services.AttrNamespace].IsSet())
+	require.True(t, config.Services[1].Metadata[services.AttrDeploymentName].IsSet())
+	_, exists := config.Services[1].Metadata[services.AttrDaemonSetName]
+	require.False(t, exists)
+	require.True(t, config.Services[2].Metadata[services.AttrNamespace].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrPodName].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrDeploymentName].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrReplicaSetName].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrStatefulSetName].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrDaemonSetName].IsSet())
+	require.True(t, config.Services[2].Metadata[services.AttrOwnerName].IsSet())
+	require.True(t, config.Services[2].PodLabels["test"].IsSet())
 	require.NoError(t, config.Services.Validate())
+	require.Len(t, config.ExcludeServices, 1)
+	require.Equal(t, "test", config.ExcludeServices[0].Name)
+	require.Equal(t, "default", config.ExcludeServices[0].Namespace)
+}
+
+func TestConvert_Prometheus(t *testing.T) {
+	args := Prometheus{
+		Features:         []string{"application", "network"},
+		Instrumentations: []string{"redis", "sql"},
+	}
```

Comment:
we need the sad paths covered as well. What happens if I set a field on features or on instrumentations that does not exist?

Reply:
Adding validation code and tests for both